test/features/caching/mailbox_cache_client_test.dart

```diff
@@ -0,0 +1,181 @@
+import 'dart:io';
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:tmail_ui_user/features/caching/config/hive_cache_config.dart';
+import 'package:tmail_ui_user/features/caching/mailbox_cache_client.dart';
+import 'package:tmail_ui_user/features/mailbox/data/extensions/mailbox_extension.dart';
+
+import '../../fixtures/mailbox_fixtures.dart';
+
+void main() {
+
+  late MailboxCacheClient _mailboxCacheClient;
+
+  setUpAll(() {
+    HiveCacheConfig().setUp(cachePath: Directory.current.path);
+  });
+
+  setUp(() {
+    _mailboxCacheClient = MailboxCacheClient();
+  });
+
+  group('[delete]', () {
+    test('cache should delete item successfully when cache empty', () async {
+      await _mailboxCacheClient.deleteItem(MailboxFixtures.inboxMailbox.id.toString());
```

Comment:
This is a bit absurd. There is no data in the database so you can delete it and then compare again. I think in this case you are always right to leave this out,